ddl/ddl_api.go

```diff
@@ -334,7 +427,7 @@ func checkMultiSchemaSpecs(sctx sessionctx.Context, specs []*ast.DatabaseOption)
 	return nil
 }
 
-func (d *ddl) AlterSchema(ctx sessionctx.Context, stmt *ast.AlterDatabaseStmt) (err error) {
+func (d *ddl) AlterSchema(ctx context.Context, sctx sessionctx.Context, stmt *ast.AlterDatabaseStmt) (err error) {
```

Comment:
Did you use context? If not, maybe just remove this argument.

Reply:
Explained in execWithTimeout

Reply:
If it is only introduced for tests, maybe just avoid it. Context from executor are mostly `context.TODO()` or server. So it means, the ddl job may be interrupted by server connection killing, which is weird for DDL.

You could use failpoint for testing instead.

Reply:
Fixed by failpoint